packages/gatsby-transformer-documentationjs/src/gatsby-node.js

```diff
@@ -17,49 +11,35 @@ const stringifyMarkdownAST = (node = ``) => {
   }
 }
 
-const commentId = (parentId, commentNumber) =>
-  `documentationJS ${parentId} comment #${commentNumber}`
+const docId = (parentId, docsJson) =>
+  `documentationJS ${parentId} path #${JSON.stringify(docsJson.path)}`
 const descriptionId = (parentId, name) =>
   `${parentId}--DocumentationJSComponentDescription--${name}`
 
-function createDescriptionNode(
-  node,
-  docNodeId,
-  markdownStr,
-  name,
-  actions,
-  createNodeId
-) {
-  const { createNode } = actions
+function prepareDescriptionNode(node, markdownStr, name, helpers) {
+  const { createNodeId, createContentDigest } = helpers
 
   const descriptionNode = {
-    id: createNodeId(descriptionId(docNodeId, name)),
+    id: createNodeId(descriptionId(node.id, name)),
     parent: node.id,
     children: [],
     internal: {
       type: `DocumentationJSComponentDescription`,
       mediaType: `text/markdown`,
       content: markdownStr,
-      contentDigest: digest(markdownStr),
+      contentDigest: createContentDigest(markdownStr),
     },
   }
 
-  node.children = node.children.concat([descriptionNode.id])
-  createNode(descriptionNode)
-
-  return descriptionNode.id
+  return descriptionNode
 }
 
 /**
  * Implement the onCreateNode API to create documentation.js nodes
  * @param {Object} super this is a super param
  */
-exports.onCreateNode = async ({
-  node,
-  loadNodeContent,
-  actions,
-  createNodeId,
-}) => {
+exports.onCreateNode = async ({ node, actions, ...helpers }) => {
+  const { createNodeId, createContentDigest } = helpers
```

Comment:
here it makes more actual sense, because I can easily pass `helpers` later to `prepareDescriptionNode` without reconstructing `helpers` object. Fact that I grab `...helpers` just future-proofs the code a bit, that if `prepareDescriptionNode` need access to another helper - it already has it without the need to adjust calls.